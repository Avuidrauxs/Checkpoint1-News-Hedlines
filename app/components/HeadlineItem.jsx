import React from 'react';
import {connect} from 'react-redux';
import { fetchAllArticles } from 'actions';

export class HeadlineItem extends React.Component{

  constructor(props){
    super(props)

  }
  handleSubmit(e){
    e.preventDefault();
    const { Top } = this.refs;
    const{ dispatch, urlText, onClickURL, id } = this.props;
    //const urlText = this.props.url;

    onClickURL(id);
    if(Top.checked){
    dispatch(fetchAllArticles(id.toLowerCase(), Top.checked));
    Top.checked = false;
  } else{
    dispatch(fetchAllArticles(id.toLowerCase(), false));
  }
}
//  <a className= 'button' href={url} target='blank' ref='url'>More Info</a>

  render(){
    const { name, description, url, dispatch } = this.props;

    return (
      <div>
        <form onSubmit={this.handleSubmit.bind(this)}>
          <h4>{name}</h4>
          <p>{description}.</p>


          <button className='button' >Read More</button>
            <label>
            <input type='checkbox' ref='Top' onChange={this.handleSubmit.bind(this)} />
            Latest Headlines
            </label>
        </form>
      </div>
    )
  }

}
export default connect()(HeadlineItem)
