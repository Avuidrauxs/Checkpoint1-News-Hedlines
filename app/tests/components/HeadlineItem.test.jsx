import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-addons-test-utils';
import expect from 'expect';


import HeadlineItem from '../../components/HeadlineItem';

describe('Headline Item', () => {
  it('should exist', () => {
    expect(HeadlineItem).toExist();
  });
  it('renders without crashing', () => {
       const div = document.createElement('div');
       ReactDOM.render(<HeadlineItem/>, div);
   });
});
