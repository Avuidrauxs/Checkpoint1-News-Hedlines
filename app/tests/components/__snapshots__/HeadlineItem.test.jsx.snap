// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline Item should render a snapshot 1`] = `
<div>
  <h4>
    
  </h4>
  <p>
    
    .
  </p>
  <button
    className="button"
    onClick={[Function]}
  >
    View Articles
  </button>
  <div>
    <group
      className="inline-radio"
    />
  </div>
</div>
`;
